lib/net.js

```diff
@@ -2118,19 +2119,38 @@ Server.prototype.listen = function(...args) {
   throw new ERR_INVALID_ARG_VALUE('options', options);
 };
 
+function isIpv6Loopback({ address, family }) {
+  return family === 6 && StringPrototypeStartsWith(address, 'fe80::');
+}
+
+function filterOnlyValidAddress(addresses) {
+  // Return the first non IPV6 loopback address if present
+  for (const address of addresses) {
+    if (!isIpv6Loopback(address)) {
+      return address;
+    }
+  }
+
+  // Otherwise return the first address
+  return addresses[0];
+}
+
 function lookupAndListen(self, port, address, backlog,
                          exclusive, flags) {
   if (dns === undefined) dns = require('dns');
   const listeningId = self._listeningId;
-  dns.lookup(address, function doListen(err, ip, addressType) {
+
+  dns.lookup(address, { all: true }, (err, addresses) => {
     if (listeningId !== self._listeningId) {
       return;
     }
     if (err) {
       self.emit('error', err);
     } else {
-      addressType = ip ? addressType : 4;
-      listenInCluster(self, ip, port, addressType,
+      const validAddress = filterOnlyValidAddress(addresses);
+      const family = validAddress?.family || 4;
+
+      listenInCluster(self, validAddress.address, port, family,
```

Comment:
My understanding is that `addresses` could be `[]`, in which case `validAddress` would be undefined, so this needs a `?` (or we'll crash in that case). Is that right? If not the preceding line needs the `?` removing instead.

Reply:
@pimterry I don't think this is valid because when the resolution fails, the error `ENOTFOUND` would be thrown